soccer/gameplay/GameplayModule.cpp

```diff
@@ -8,22 +8,47 @@
 
 #include <stdio.h>
 #include <iostream>
+#include <cmath>
 
 // for python stuff
 #include "robocup-py.hpp"
 
+using namespace Gameplay;
+
+REGISTER_CONFIGURABLE(GameplayModule);
+
 using namespace std;
 using namespace boost;
 using namespace boost::python;
 
 using namespace Geometry2d;
 
+ConfigDouble* GameplayModule::_fieldEdgeInset;
+
+void GameplayModule::createConfiguration(Configuration* cfg) {
+    // this sets the disance from the field boundries to the edge of the global
+    // obstacles, which the
+    // robots will not move through or into
+    _fieldEdgeInset = new ConfigDouble(cfg, "Field Edge Obstacle", .3);
+}
+
+const bool GameplayModule::hasFieldEdgeInsetChanged() {
+    if (abs(_fieldEdgeInset->value() - _oldFieldEdgeInset) >
+        numeric_limits<double>::epsilon()) {
+        _oldFieldEdgeInset = _fieldEdgeInset->value();
```

Comment:
Setting the value of `_oldFieldEdgeInset` should probably be at the end of `calculateFieldObstacles()` instead of here.  As it is now, if the config value changes, a call to `hasFieldEdgeInsetChanged()` will return true, but another call right afterwards will return false.  It would be better if it returned true until the obstacles have been updated.